src/coinify/trade.js

```diff
@@ -164,10 +164,15 @@ CoinifyTrade.prototype.set = function (obj) {
 };
 
 CoinifyTrade.prototype.removeLabeledAddress = function () {
-  var account;
-  if (this._account_index && Helpers.isPositiveInteger(this._receive_index)) {
-    account = MyWallet.wallet.hdwallet.accounts[this._account_index];
-    account.removeLabelForReceivingAddress(this._receive_index);
+  if (Helpers.isPositiveInteger(this._account_index) && Helpers.isPositiveInteger(this._receive_index)) {
+    var account = MyWallet.wallet.hdwallet.accounts[this._account_index];
+    var currentLabel = account.getLabelForReceivingAddress(this._receive_index);
+    if (currentLabel === 'Coinify order #' + this.id) {
```

Comment:
This means that if the user changes the label, it won't get deleted. It's also going to break if we ever localize that string, or if we change the string between versions.

Instead of looking at the labels, it's probably safer to iterate through coinify.trades and check if any other trade uses the same account + receive index. Then simply rewrite the label.